install.sh

```diff
@@ -161,28 +141,129 @@ __atuin_install_unsupported(){
 		esac
 	done
 }
+__atuin_check_gcc(){
+    OS=$(__get_os)
+    if command -v gcc &> /dev/null; then
+        echo "gcc is installed"
+    else
+        echo "gcc not installed. Please install gcc."
+        exit
+    fi
+}
+
+__get_os() {
+  if [ -f "/etc/os-release" ]; then
+    OS="$(grep -Po '(?<=^ID=).*$' /etc/os-release | tr '[:upper:]' '[:lower:]')" 2>/dev/null
+  elif [ -f "/etc/debian_version" ]; then
+    OS="debian"
+  elif [ -f "/etc/centos-release" ]; then
+    OS="centos"
+  elif [ -f "/etc/manjaro-release" ]; then
+    OS="manjarolinux"
+  elif [ -f "/etc/fedora-release" ]; then
+    OS="fedora"
+    __atuin_check_gcc
+  elif [ -f "/etc/arch-release" ]; then
+    OS="arch"
+  elif [ -f "/etc/gentoo-release" ]; then
+    OS="gentoo"
+  else
+    if uname -r | grep -qi "microsoft"; then
+      if [ -f "/etc/os-release" ]; then
+        PRETTY_NAME="$(grep -Po '(?<=^PRETTY_NAME=).*$' /etc/os-release | tr -d '"' | tr '[:upper:]' '[:lower:]')"
+        case "$PRETTY_NAME" in
+          "ubuntu"* ) OS="ubuntu" ;;
+          "debian"* ) OS="debian" ;;
+          "fedora"* ) OS="fedora" ;;
+          # Add more distributions as needed
+          * ) OS="unknownwsl" ;;
+        esac
+      else
+        OS="unknownwsl"
+      fi
+    else
+      echo "Unknown"
```

Comment:
We get to this case running on macOS

Reply:
Small mistake with the get OS function, apologies, now fixed! 

```
  if [[ "$OSTYPE" == "darwin"* ]]; then
    echo "darwin"
  elif [ -f "/etc/os-release" ]; then
```